base/promotion.jl

```diff
@@ -222,9 +222,8 @@ minmax(x::Real, y::Real) = minmax(promote(x, y)...)
 # for the multiplication of two types,
 #   promote_op{R<:MyType,S<:MyType}(::typeof(*), ::Type{R}, ::Type{S}) = MyType{multype(R,S)}
 promote_op(::Any)    = (@_pure_meta; Bottom)
-promote_op(::Any, T) = (@_pure_meta; T)
-promote_op{T}(::Type{T}, ::Any) = (@_pure_meta; T)
```

Comment:
Why not? Sounds like a case where you might want to store non-`Int` values in the resulting array:
```julia
julia> x = Real.([1+0im])
1-element Array{Real,1}:
 1

julia> x[1] = 1.5
1.5
```